modules/network/kademlia/kademlia-utils.js

```diff
@@ -78,15 +79,104 @@ class KademliaUtils {
     }
 
     /**
-    * Verifies if we are on the test network and otherconfig checks
-    */
+     * Verifies if we are on the test network and otherconfig checks
+     */
     verifyConfiguration(config) {
         if (config.traverse_nat_enabled && config.onion_enabled) {
             this.log.error('Refusing to start with both TraverseNatEnabled and ' +
-          'OnionEnabled - this is a privacy risk');
+                'OnionEnabled - this is a privacy risk');
             process.exit(1);
         }
     }
+
+    /**
+     * Save bootstrap nodes
+     */
+    setBootstraps(nodes) {
+        const filePath = path.join(this.config.appDataPath, 'bootstraps.json');
+        if (fs.existsSync(filePath)) {
+            fs.unlinkSync(filePath);
+        }
+
+        fs.writeFileSync(filePath, JSON.stringify(nodes));
+    }
+
+    /**
+     * Check if bootstrap nodes have changed
+     */
+    checkBootstraps(nodes) {
+        const filePath = path.join(this.config.appDataPath, 'bootstraps.json');
+        if (!fs.existsSync(filePath) || nodes.length === 0) {
+            return false;
+        }
+
+        const existingBootstraps = JSON.parse(fs.readFileSync(filePath));
+
+        for (let i = 0; i < existingBootstraps.length; i += 1) {
+            if (nodes.indexOf(existingBootstraps[i]) === -1) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    /**
+     * Save routing table
+     */
+    setRoutingTable(map) {
+        const obj = {};
+
+        map.forEach((value, key) => {
+            const childObj = {};
+            value.forEach((childValue, childKey) => {
+                childObj[childKey] = childValue;
+            });
+            obj[key] = childObj;
+        });
+
+        const filePath = path.join(this.config.appDataPath, 'router.json');
+        if (fs.existsSync(filePath)) {
+            fs.unlinkSync(filePath);
+        }
+
+        const result = JSON.stringify({ timestamp: new Date(), routingTable: obj });
+        fs.writeFileSync(filePath, result);
+    }
+
+    /**
+     * Load routing table
+     */
+    getRoutingTable(router) {
+        const routinTableValidityPeriod =
+            1000 * 60 * 60 * this.config.network.routing_table_validity_period_in_hours;
+        const filePath = path.join(this.config.appDataPath, 'router.json');
+        if (!fs.existsSync(filePath)) {
+            return false;
+        }
+        const obj = JSON.parse(fs.readFileSync(filePath));
+        const { routingTable } = obj;
```

Comment:
This should probably be inside the try block just in case it throws an error for a non JSON or an empty file